lib/cudadrv/occupancy.jl

```diff
@@ -36,7 +36,13 @@ end
 # HACK: callback function for `launch_configuration` on platforms without support for
 #       trampolines as used by `@cfunction` (JuliaLang/julia#27174, JuliaLang/julia#32154)
 _shmem_cb = nothing
-_shmem_cint_cb(x::Cint) = Cint(something(_shmem_cb)(x))
+function _shmem_cint_cb(x::Cint)
+    # see @gcsafe_ccall documentation
+    @static if VERSION < v"1.9"
+        GC.safepoint()
```

Comment:
Did you mean:

```
state = @ccall jl_gc_unsafe_enter()::Int8
```

and at the end:

```
@ccall jl_gc_unsafe_leave(state::Int8)
```

Reply:
I didn't, but I guess I may be misunderstanding. I thought that acquiring a safepoint when GC is active would block / be sufficient of a workaround, because you mentioned that manual action isn't needed on 1.10 now that we have safepoint_on_entry.

Reply:
I would use:
```
macro gcunsafe(expr)
    if VERSION < v"1.9.0"
        quote
            state = ccall(:jl_gc_unsafe_enter, Int8, ())
            val = $(esc(expr))
            ccall(:jl_gc_unsafe_leave, Cvoid, (Int8,), state)
            val
        end
    else
        return expr
    end
end
```

To wrap the callback functions. I can push that update?

Reply:
> GC could start immediatly after the safepoint and we would have the same issue.

Ah right, that's the crucial bit I was missing. Thanks!

Reply:
I'll push the update, as I'll have to rebase this branch on top of the header regenerate PR anyway.